source/dub/test/subpackages.d

```diff
@@ -38,3 +38,20 @@ unittest
 
     assert(!dub.packageManager().getPackage(PackageName("b:b"), Version("1.1.0")));
 }
+
+// https://github.com/dlang/dub/issues/2973
+unittest
+{
+    scope dub = new TestDub((scope Filesystem root) {
+        root.writeFile(TestDub.ProjectPath ~ "dub.json",
+            `{ "name": "a", "dependencies": { "b:a": "~>1.0" } }`);
+        root.writeFile(TestDub.ProjectPath ~ "dub.selections.json",
+            `{ "fileVersion": 1, "versions": { "b": "1.0.0" } }`);
+        root.writePackageFile("b", "1.0.0",
+            `{ "name": "b", "version": "1.0.0", "subPackages": [ { "name": "a" } ] }`);
+    });
```

Comment:
Btw thx again for the test framework, sooo much better than the old tests...

Reply:
I wish we had a similar filesystem abstract in Phobos to make this kind of unittests the norm. It is so much more expressive.
I really need to finish it though, there's an issue with Windows at the moment which prevents mimicking the real world properly.